packages/core/admin/admin/src/pages/Webhooks/ListView/index.js

```diff
@@ -276,48 +247,183 @@ function ListView() {
   }
 
   return (
-    <Wrapper>
+    <Layout>
       <PageTitle name="Webhooks" />
-      <Header {...headerProps} />
-      {canRead && (
-        <div className="list-wrapper">
+      <HeaderLayout
+        title={headerProps.title.label}
+        subtitle={headerProps.content}
+        primaryAction={(
+          <Button onClick={() => (canCreate ? handleGoTo('create') : {})} startIcon={<AddIcon />}>
+            {headerProps.addBtnLabel}
+          </Button>
+        )}
+      />
+
+      <ContentLayout>
+        <>
           {rowsCount > 0 ? (
-            <List
-              {...listProps}
-              customRowComponent={props => {
-                return (
-                  <ListRow
-                    {...props}
-                    canUpdate={canUpdate}
-                    canDelete={canDelete}
-                    onCheckChange={handleChange}
-                    onEditClick={handleGoTo}
-                    onDeleteCLick={handleDeleteClick}
-                    onEnabledChange={handleEnabledChange}
-                    itemsToDelete={webhooksToDelete}
-                  />
-                );
-              }}
-            />
+            <Table
+              colCount={5}
+              rowCount={webhooks.length}
+              footer={(
+                <TFooter onClick={() => (canCreate ? handleGoTo('create') : {})} icon={<AddIcon />}>
+                  Add another field to this collection type
+                </TFooter>
+              )}
+            >
+              <Thead>
+                <Tr>
+                  <Th>
+                    <BaseCheckbox
+                      aria-label="Select all entries"
+                      indeterminate={
+                        webhooksToDelete.length > 0 && webhooksToDelete.length < webhooks.length
+                      }
+                      value={webhooksToDelete.length === webhooks.length}
+                      onValueChange={value => handleCheckboxChange('main', value)}
```

Comment:
I would suggest to separate the logic between checking the "main" checkbox and checking the other ones. This will reduce the complexity of the `handleCheckboxChange`. Complexity, in this scenario, corresponds to the number paths existing in the "handleCheckboxChange". 

And for clarity reasons, I would suggest to avoid the conditional in this function since it's arbitrary and note based on the data we have 😊 

Something like:

```js
const handleSelectAllCheckbox = () => {}

const handleSelectOneCheckbox = () => {}
```